Src/HALAL/Services/Communication/I2C/I2C.cpp

```diff
@@ -0,0 +1,248 @@
+/*
+ * SPI.hpp
+ *
+ *  Created on: 5 nov. 2022
+ *      Author: Pablo
+ */
+#include "Communication/I2C/I2C.hpp"
+
+#ifdef HAL_I2C_MODULE_ENABLED
+
+unordered_map<uint8_t, I2C::Instance* > I2C::active_i2c;
+
+uint16_t I2C::id_counter = 0;
+
+optional<uint8_t> I2C::inscribe(I2C::Peripheral& i2c){
+	if ( !I2C::available_i2cs.contains(i2c)){
+
+		return nullopt; //TODO: Error handler or throw the exception
+	}
+
+	I2C::Instance* i2c_instance = I2C::available_i2cs[i2c];
+
+    Pin::inscribe(i2c_instance->SCL, ALTERNATIVE);
+    Pin::inscribe(i2c_instance->SDA, ALTERNATIVE);
+
+    DMA::inscribe_stream(i2c_instance->RX_DMA);
+    DMA::inscribe_stream(i2c_instance->TX_DMA);
+
+    uint8_t id = I2C::id_counter++;
+
+    I2C::active_i2c[id] = i2c_instance;
+
+    return id;
+}
+
+
+void I2C::start(){
+	for(auto iter: I2C::active_i2c){
+		I2C::init(iter.second);
+	}
+}
+
+
+bool I2C::transmit_next_packet(uint8_t id, I2CPacket& packet){
+    if (!I2C::active_i2c.contains(id)){
+    	ErrorHandler("I2C No encontrado al enviar paquete \n\r");
+        return false; //TODO: Error handler
```

Comment:
```suggestion
        return false; //TODO: Error handler
```
Añadir Error Handler